src/inspector/models/score/scoreaccessibilitysettingsmodel.cpp

```diff
@@ -0,0 +1,169 @@
+    switch (preset) {
+    case ScoreStylePreset::DEFAULT:
+        return engravingConfiguration()->defaultStyleFilePath();
+    case ScoreStylePreset::MSN_16MM:
+        return basePath + "16mm_MSN.mss";
+    case ScoreStylePreset::MSN_18MM:
+        return basePath + "18mm_MSN.mss";
+    case ScoreStylePreset::MSN_20MM:
+        return basePath + "20mm_MSN.mss";
+    case ScoreStylePreset::MSN_22MM:
+        return basePath + "22mm_MSN.mss";
+    case ScoreStylePreset::MSN_25MM:
+        return basePath + "25mm_MSN.mss";
+    default:
+        return muse::io::path_t();
+    }
+}
+
+QVariantList ScoreAccessibilitySettingsModel::possibleScoreStylePreset() const
+{
+    QVariantList presets;
+    MStyle style = globalContext()->currentNotation()->elements()->msScore()->score()->style();
+    int currentPresetValue = static_cast<int>(style.stylePreset());
+
+    for (int i = 0; i < static_cast<int>(ScoreStylePreset::MAX_PRESET); ++i) {
+        ScoreStylePreset presetEnum = static_cast<ScoreStylePreset>(i);
+        QString text = TConv::translatedUserName(presetEnum);
+
+        QVariantMap presetMap = {
+            { "text", text },
+            {
+                "value",
+                QVariantMap {
+                    { "preset", i },
+                    { "edited", false }
+                }
+            }
+        };
+        presets.append(presetMap);
+
+        if (i == currentPresetValue && style.stylePresetEdited()) {
+            QVariantMap editedPresetMap = presetMap;
+            editedPresetMap["text"] = muse::qtrc("inspector", "%1 (edited)").arg(text);
+            editedPresetMap["value"].toMap()["edited"] = true;
+            presets.append(editedPresetMap);
+        }
+    }
+
+    return presets;
+}
+
+void ScoreAccessibilitySettingsModel::loadStyle(int preset)
+{
+    ScoreStylePreset presetName = static_cast<ScoreStylePreset>(preset);
+    muse::io::path_t filePath = getStyleFilePath(presetName);
+    m_ignoreStyleChange = true;
+
+    if (presetName == ScoreStylePreset::DEFAULT) {
+        if (filePath.empty()) {
+            StyleIdSet emptySet;
+            globalContext()->currentNotation()->style()->resetAllStyleValues(emptySet);
+        } else {
+            globalContext()->currentNotation()->style()->loadStyle(filePath, true);
+        }
+    } else if (!filePath.empty()) {
+        LOGI() << "Loading style from filePath:" << filePath;
+        globalContext()->currentNotation()->style()->loadStyle(filePath, true);
+    } else {
+        LOGI() << "filePath is empty";
+    }
+    globalContext()->currentNotation()->elements()->msScore()->score()->style().setStylePreset(presetName);
+
+    isAccessibilityStyleEdited = false;
+    globalContext()->currentNotation()->elements()->msScore()->score()->style().setStylePresetEdited(false);
+    emit accessibilityStyleEditedChanged();
+    m_ignoreStyleChange = false;
+}
+
+int ScoreAccessibilitySettingsModel::scoreStylePreset() const
+{
```

Comment:
This function should return the value directly rather than an index to the value.
```C++
return QVariantMap {
    { "preset", style.stylePreset() },
    { "edited", style.stylePresetEdited() }
}
```

Then in `ScoreAccessibilityInspectorView.qml` you can do:

```QML
scoreStylePreset.currentIndex = indexOfValue(root.model.scoreStylePreset())
```

Reply:
@shoogle I tried to do this but came across a finding that there is strict check in the `indexOfValue()` method [here](https://github.com/musescore/MuseScore/blob/f1a01b2ecfd93542b93cf10a0de2f51b87bc323c/src/framework/uicomponents/qml/Muse/UiComponents/StyledDropdown.qml#L64). So as per [JS Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) `If both operands are objects, return true only if they refer to the same object.` which is not the case here due to the fact that even though two objects have the same values, they are considered different objects in memory unless they are the exact same reference so it was going back to -1. I thought of multiple solutions but the one where we pass the model to the `scoreStylePreset()` sounded fair. Ideally, this method should now be called like "getScoreStylePresetIndex()" or something which would look like ```
QVariantMap ScoreAccessibilitySettingsModel::getScoreStylePresetIndex(const QVariantList& presets) const
{
    MStyle style = globalContext()->currentNotation()->elements()->msScore()->style();
    int currentPresetValue = static_cast<int>(style.stylePreset());

    for (const QVariant& preset : presets) {
        QVariantMap presetMap = preset.toMap();
        QVariantMap valueMap = presetMap.value("value").toMap();
        int presetValue = valueMap.value("preset").toInt();
        bool editedValue = valueMap.value("edited").toBool();

        if (presetValue == currentPresetValue && editedValue == style.stylePresetEdited()) {
            return preset.toMap().value("value").toMap();
        }
    }
    
    return presets.first().toMap().value("value").toMap(); // or we can do: return QVariantMap();
}
``` I tested this approach and it works. Let me know your thoughts on this?

Reply:
@shoogle I tried to do this but came across a finding that there is [strict check](https://github.com/musescore/MuseScore/blob/f1a01b2ecfd93542b93cf10a0de2f51b87bc323c/src/framework/uicomponents/qml/Muse/UiComponents/StyledDropdown.qml#L64) in the `indexOfValue()` method. So as per [JS Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) `If both operands are objects, return true only if they refer to the same object.` which is not the case here due to the fact that even though two objects have the same values, they are considered different objects in memory unless they are the exact same reference so it was going back to `-1` index. I thought of multiple solutions but the one where we pass the model to the `scoreStylePreset()` sounded fair. Ideally, this method should now be called like `getScoreStylePresetIndex()` or something which would look like 

```
QVariantMap ScoreAccessibilitySettingsModel::getScoreStylePresetIndex(const QVariantList& presets) const
{
    MStyle style = globalContext()->currentNotation()->elements()->msScore()->style();
    int currentPresetValue = static_cast<int>(style.stylePreset());

    QString text = "text";
    QString value = "value";
    QString preset = "preset";
    QString edited = "edited";

    for (const QVariant& p : presets) {
        QVariantMap presetMap = p.toMap();
        QVariantMap valueMap = presetMap.value(value).toMap();
        int presetValue = valueMap.value(preset).toInt();
        bool editedValue = valueMap.value(edited).toBool();

        if (presetValue == currentPresetValue && editedValue == style.stylePresetEdited()) {
            return p.toMap().value(value).toMap();
        }
    }
    
    return presets.first().toMap().value(value).toMap(); // or we could do: return QVariantMap();
}
``` or Am I missing something? I checked when I do `JSON.stringify()` in the JS file, it works

Reply:
I think it's okay for this method to return `int`, but would rename it to `currentScoreStylePresetIndex`. However the implementation can be cleaned up; casting to `int` and then checking every possible `int` value is not great. Either rewrite it as a switch, without a cast to `int`; or as an arithmetic expression, i.e. `return isAccessibilityStyleEdited ? presetIndex + 1 : presetIndex`.

Reply:
Ahh, good catch! I actually had a go at this myself and wondered why it wasn't working... 🤪

There are three options here:

1. Make the [check](https://github.com/musescore/MuseScore/blob/f1a01b2ecfd93542b93cf10a0de2f51b87bc323c/src/framework/uicomponents/qml/Muse/UiComponents/StyledDropdown.qml#L64) non-strict (i.e. use `==` instead of `===`).
    - Qt Creator will display a warning if you do this, so you'd need to leave a comment there so that nobody "fixes" it later on.
2. Unpack the `possibleScoreStylePresets()` list and return the actual object rather than an equivalent one.
    - I don't think this would work though, because you'd get a different (albeit equivalent) object each time you call the function.
    - If it happens to always give the same object on your machine, that might be a compiler optimization that wouldn't necessarily work with other compilers or with other compiler settings.
4. Return an index instead of a value.

Your suggested code is a strange mix of 2 and 3. You rename the function to include "index", but then you return an object (i.e. a value) from the list.

Perhaps it is best to return an index after all. If so, this is the way to do it:

```C++
int ScoreAccessibilitySettingsModel::scoreStylePresetIndex() const
{
    MStyle style = globalContext()->currentNotation()->elements()->msScore()->style();
    int currentPresetValue = static_cast<int>(style.stylePreset());
    if (style.stylePresetEdited()) {
        return currentPresetValue + 1;
    }
    return currentPresetValue;
}
```

This works because when you [looped over the enum values](https://github.com/musescore/MuseScore/pull/23048/files#diff-68cac91f52e04f14d17e9fa1928b497c8f5aaf9cf08a2ea8fb3006807b404e7eR93) to create the `QVariantList` in `possibleScoreStylePresets()`, you already assumed that the first enum value is 0 and it increments by 1 each time. 

If that assumption is true (which we know it is from `types.h`) then it follows that the index in the list is equal to the enum value. If it's edited you just need to add 1 to that value.

The presence of the edited value in the list throws all the later indexes off by one. But the neat thing is that the edited one is removed from the list as soon as it is no longer the current one, so this function never needs to return an index "after" the edited one.

Reply:
TLDR: Pick option 3 and use that code snippet. It will be faster than the other options because it avoids iterating over the list. You can use your knowledge of the enum to jump straight to the correct index.

Reply:
Ah yeah this makes sense and much much cleaner to what I was doing. Just updated the PR